plugins/database/mysql/connection_producer.go

```diff
@@ -142,12 +181,22 @@ func (c *mySQLConnectionProducer) Connection(ctx context.Context) (interface{},
 		c.db.Close()
 	}
 
+	driverName := driverMySQL
+	if c.isCloud {
```

Comment:
nit: could we check if `cloudDriverName` is set instead of also maintaining a bool flag?

Reply:
we can - i was originally piggybacking on the TLSConfig string, which would have necessitated the flag

Reply:
In the case of  TLSConfig, we can't use that because sometimes we want to enable CloudSQL without TLS? Is that right?

Reply:
yeah, the two are orthogonal, so if the string was non-empty, we'd still need a way to know which thing (cloud/tls) was active